haproxy_exporter.go

```diff
@@ -424,15 +435,18 @@ func (e *Exporter) parseRow(csvRow []string, ch chan<- prometheus.Metric) {
 	case backend:
 		e.exportCsvFields(backendMetrics, csvRow, ch, pxname)
 	case server:
-		e.exportCsvFields(e.serverMetrics, csvRow, ch, pxname, svname)
+
+		if _, ok := e.excludedServerStates[status]; !ok {
+			e.exportCsvFields(e.serverMetrics, csvRow, ch, pxname, svname)
+		}
 	}
 }
 
 func parseStatusField(value string) int64 {
 	switch value {
-	case "UP", "UP 1/3", "UP 2/3", "OPEN", "no check":
+	case "UP", "UP 1/3", "UP 2/3", "OPEN", "no check", "DRAIN":
 		return 1
-	case "DOWN", "DOWN 1/2", "NOLB", "MAINT":
+	default: //case "DOWN", "DOWN 1/2", "NOLB", "MAINT", "MAINT(via)", "MAINT(resolution)":
```

Comment:
We already return 0 / down as a fallback. What's the motivation for this change instead of listing known down status explicitly?

Reply:
I wasn't sure what to do with this function, since I haven't found a complete list of possible servers statuses. I considered extending the list with the statuses I knew about, but decided there was not point, especially since (as you say) the fallback was 0 / down as fallback anyway.